docs/best-practices.md

```diff
@@ -0,0 +1,134 @@
+# Best practices for writing actors
+
+## Follow the coding guidelines
+
+See the [Python coding guidelines for leapp-repository](CONTRIBUTING.md#coding-guidelines).
+
+## Avoid running code on a module level
+
+Despite the actors being written in Python, they are loaded beforehand to get all the meta-information from them. To
+avoid slow downs during this load phase we ask you to not execute any code that is not absolutely necessary
+on a module level. Simple, fast code with no side-effects is acceptable, for example constant definitions.
+
+## Avoid certain global imports
+
+On a module level, try to import only the Python Standard Library modules or modules provided by the Leapp framework.
+Import any other module within your function or method. This has to do with possible slow down when loading repositories
+by the leapp tool, and also to avoid unnecessary complications for our tests automation which needs to inspect the
+actors beforehand.
+
+## Use the snactor tool for development
+
+The snactor tool helps you with creating the base layout of a new Leapp repository, and with creating boilerplates for
+the repository artifacts like actors, models, tags, topics, and workflows. It also helps you with debugging as it is
+able to execute individual actors.
+
+See the [tutorial on basic usage of snactor](https://leapp.readthedocs.io/en/latest/first-actor.html).
+
+## Avoid writing generic actors
```

Comment:
I would modify the title as writting of generic actors is important too :) The content is not about generic acotrs, but about generic functionality inside the actor. From this point, the titleis confusing.
- Michal's idea: Move generic functonality to libraries

Reply:
Changed per my idea :)